File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """

        list3=[]

        for num in nums1:
            if num in nums2:
                list3.append(num)
                nums2.remove(num)
        return list3

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        

         int[] frequencyArray = new int[1001];

         for(int num : nums1)
         {
            frequencyArray[num]++;
         }

         ArrayList<Integer> list = new ArrayList<>();
         for(int num:nums2)
         {
            if(frequencyArray[num]>0)
            {
                list.add(num);
                frequencyArray[num]--;
            }
         }

         int[] res = new int[list.size()];

         for(int i=0;i<list.size();i++)
         {
            res[i] = list.get(i);
         }
        return res;
    }
}